src/workloads/execution/ExpressiveQueries.yml

```diff
@@ -0,0 +1,175 @@
+        zip: {^FastRandomString: {length: 6, alphabet: "0123456789"}},
+        location: {position: {type: "Point", coordinates: [
+          {^RandomDouble: {min: -118.668469, max: -82.062023}},
+          {^RandomDouble: {min: 32.924436, max: 41.679986}}
+        ]}}
+      }
+      policies: {^Array: {number: {^RandomInt: {min: 3, max: 6}}, of: {^Choose: {from: [
+        {
+          policyType: "auto",
+          policyNum: &policyNum {^FastRandomString: {length: 30}},
+          nextRenewalDt: &nextRenewalDt {^RandomDate: {min: "2017-01-01", max: "2017-12-31"}},
+          year: {^RandomInt: {min: 1980, max: 2017}},
+          value: {^RandomInt: {min: 500, max: 500000}},
+          model: {^FastRandomString: {length: 30}},
+        },
+        {
+          policyType: "home",
+          policyNum: *policyNum,
+          nextRenewalDt: *nextRenewalDt,
+          year: {^RandomInt: {min: 1970, max: 2017}},
+          value: {^RandomInt: {min: 500, max: 10000000}},
+          address: *address
+        },
+        {
+          policyType: "life",
+          policyNum: *policyNum,
+          nextRenewalDt: *nextRenewalDt,
+          insured_person: {
+            firstname: *firstname,
+            lastname: *lastname,
+            yob: {^RandomDate: {min: "1930-01-01", max: "2016-12-31"}},
+            smoking: {^RandomInt: {min: 0, max: 1}}
+          }
+        }
+      ]}}}}
+  - &Nop {Nop: true}
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *DB
+  Phases:
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - *Nop
+  - *Nop
+  - Repeat: 1
+  - *Nop
+
+- Name: CreateIndex
+  Type: RunCommand
+  Threads: 1
+  Phases:
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  # Create index *after* running some queries without it.
+  - Repeat: 1
+    Database: *DB
+    Operations:
+    - OperationName: RunCommand
+      OperationCommand:
+        createIndexes: Collection0
+        indexes:
+        - key:
+            address.state: 1
+            policies.policyType: 1
+            policies.insured_person.smoking: 1
+            gender: 1
+            dob: 1
+          name: usefulIndex
+  - *Nop
+  - *Nop
+
+- Name: RunQueries
+  Type: RunCommand
+  Threads: 1
+  Phases:
+  - *Nop
+  - *Nop
+  - Repeat: 100
+    Duration: 5 minutes
+    Database: *DB
+    Operations:
+    - OperationMetricsName: QueryNoProjectionNoIndex
+      OperationName: RunCommand
+      OperationCommand:
+        find: Collection0
+        filter: &filter {
+          gender: "Female",
+          dob: {
+            # Currently, using generator is the only way to generate date value with the appropriate BSON type.
+            $gte: {^RandomDate: {min: "1990-01-01", max: "1990-01-01T00:00:00.001"}},
+            $lte: {^RandomDate: {min: "1990-12-31", max: "1990-12-31T00:00:00.001"}}
```

Comment:
yikes well this is gross. Can you cut a TIG ticket to support a better syntax and add a TODO here with that ticket number?

Just `{^Date:}` would be fine.  I'm also curious if `{^Verbatim: {$date:}}` (using the bson notation) would work, but not sure if the driver supports that syntax versus native types.

Reply:
Yeah this super ugly. There's already a ticket for ^Date (see PERF-2084). The driver doesn't support Extended json yet (which is a shame).